### PR TITLE
ddi: client: replace deprecated aiohttp.Timeout

### PR DESCRIPTION
**Pull request by Bastian-Krause**
Use timeout keyword for session.get(), session.put() and
session.post(). Use async-timeout's context manager to allow content
reading just like before.

Signed-off-by: Bastian Stender <bst@pengutronix.de>

### CODE REVIEW

File: rauc_hawkbit/ddi/client.py [hunk 1]
```diff
@@ -163,9 +164,9 @@ def build_api_url(self, api_path):
                     **kwargs))
 
         self.logger.debug('GET {}'.format(url))
-        with aiohttp.Timeout(self.timeout):
-            async with self.session.get(url, headers=get_headers,
-                                        params=query_params) as resp:
+        async with self.session.get(url, headers=get_headers,
+                                    params=query_params,
+                                    timeout=self.timeout) as resp:
```
Comment (ejoerns) [inline rauc_hawkbit/ddi/client.py]: It was two indentation levels before and is only one now, thus please also fix indention of the block below

File: rauc_hawkbit/ddi/client.py [hunk 2]
```diff
@@ -225,12 +226,12 @@ def build_api_url(self, api_path):
         hash_md5 = hashlib.md5()
 
         self.logger.debug('GET binary {}'.format(url))
-        with aiohttp.Timeout(timeout, loop=self.session.loop):
-            async with self.session.get(url, headers=get_bin_headers) as resp:
+        async with self.session.get(url, headers=get_bin_headers,
+                                    timeout=timeout) as resp:
```
Comment (ejoerns) [inline rauc_hawkbit/ddi/client.py]: same here as above

File: rauc_hawkbit/ddi/client.py [hunk 3]
```diff
@@ -259,9 +260,9 @@ def build_api_url(self, api_path):
                     controllerId=self.controller_id,
                     **kwargs))
         self.logger.debug('POST {}'.format(url))
-        with aiohttp.Timeout(self.timeout):
-            async with self.session.post(url, headers=post_headers,
-                                         data=json.dumps(data)) as resp:
+        async with self.session.post(url, headers=post_headers,
+                                     data=json.dumps(data),
+                                     timeout=self.timeout) as resp:
```
Comment (ejoerns) [inline rauc_hawkbit/ddi/client.py]: here

File: rauc_hawkbit/ddi/client.py [hunk 4]
```diff
@@ -285,9 +286,9 @@ def build_api_url(self, api_path):
                     **kwargs))
         self.logger.debug('PUT {}'.format(url))
         self.logger.debug(json.dumps(data))
-        with aiohttp.Timeout(self.timeout):
-            async with self.session.put(url, headers=put_headers,
-                                        data=json.dumps(data)) as resp:
+        async with self.session.put(url, headers=put_headers,
+                                    data=json.dumps(data),
+                                    timeout=self.timeout) as resp:
```
Comment (ejoerns) [inline rauc_hawkbit/ddi/client.py]: and here :)

**Review by ejoerns — CHANGES_REQUESTED** [review]
Looks fine for me, except some minor indentations that should be adjusted.

**Comment by Bastian-Krause** [comment]
Good catch, fixed.

**Comment by ejoerns** [comment]
I'll close this in favor of #8 which addressed the same issue but also enhances download speed by not limiting chunk sizes.

**Pull request closed without merging**